x-pack/plugins/security_solution/public/common/components/navigation/use_security_solution_navigation/use_navigation_items.tsx

```diff
@@ -5,62 +5,87 @@
  * 2.0.
  */
 
-import React from 'react';
+import React, { useCallback, useMemo } from 'react';
+import { EuiSideNavItemType } from '@elastic/eui/src/components/side_nav/side_nav_types';
+import { navTabGroups } from '../../../../app/home/home_navigations';
 import { APP_ID } from '../../../../../common/constants';
 import { track, METRIC_TYPE, TELEMETRY_EVENT } from '../../../lib/telemetry';
 import { getSearch } from '../helpers';
 import { PrimaryNavigationItemsProps } from './types';
-import { useKibana } from '../../../lib/kibana';
+import { useGetUserCasesPermissions, useKibana } from '../../../lib/kibana';
+import { NavTab } from '../types';
 
 export const usePrimaryNavigationItems = ({
-  filters,
   navTabs,
-  query,
-  savedQuery,
   selectedTabId,
-  sourcerer,
-  timeline,
-  timerange,
-}: PrimaryNavigationItemsProps) => {
+  ...urlStateProps
+}: PrimaryNavigationItemsProps): Array<EuiSideNavItemType<{}>> => {
   const { navigateToApp, getUrlForApp } = useKibana().services.application;
 
-  const navItems = Object.values(navTabs).map((tab) => {
-    const { id, name, disabled } = tab;
-    const isSelected = selectedTabId === id;
-    const urlSearch = getSearch(tab, {
-      filters,
-      query,
-      savedQuery,
-      sourcerer,
-      timeline,
-      timerange,
-    });
+  const getSideNav = useCallback(
+    (tab: NavTab) => {
+      const { id, name, disabled } = tab;
+      const isSelected = selectedTabId === id;
+      const urlSearch = getSearch(tab, urlStateProps);
 
-    const handleClick = (ev: React.MouseEvent) => {
-      ev.preventDefault();
-      navigateToApp(`${APP_ID}:${id}`, { path: urlSearch });
-      track(METRIC_TYPE.CLICK, `${TELEMETRY_EVENT.TAB_CLICKED}${id}`);
-    };
+      const handleClick = (ev: React.MouseEvent) => {
+        ev.preventDefault();
+        navigateToApp(APP_ID, { deepLinkId: id, path: urlSearch });
+        track(METRIC_TYPE.CLICK, `${TELEMETRY_EVENT.TAB_CLICKED}${id}`);
```

Comment:
do we still need that if we are using `<TrackApplicationView`?

Reply:
nope, not anymore. I remove it